components/accordian-item.js

```diff
@@ -3,12 +3,41 @@ import AccordionContext from 'react-bootstrap/AccordionContext'
 import { useAccordionButton } from 'react-bootstrap/AccordionButton'
 import ArrowRight from '@/svgs/arrow-right-s-fill.svg'
 import ArrowDown from '@/svgs/arrow-down-s-fill.svg'
-import { useContext } from 'react'
+import { useRouter } from 'next/router'
+import { useContext, useEffect, useState } from 'react'
+import { useFormikContext } from 'formik'
 
-function ContextAwareToggle ({ children, headerColor = 'var(--theme-grey)', eventKey }) {
+function ContextAwareToggle ({ children, headerColor = 'var(--theme-grey)', eventKey, hasForm }) {
+  const router = useRouter()
   const { activeEventKey } = useContext(AccordionContext)
+  const formik = hasForm ? useFormikContext() : null
+  const [show, setShow] = useState(undefined)
   const decoratedOnClick = useAccordionButton(eventKey)
 
+  useEffect(() => {
+    const isDirty = formik?.values.forward?.[0].nym !== '' || formik?.values.forward?.[0].pct !== '' || formik?.values.boost !== '' || (router.query?.type === 'link' && formik?.values.text !== '')
+
+    if (isDirty && hasForm) {
+      setShow(false)
+    }
+  }, [formik?.values])
+
+  useEffect(() => {
+    const hasBoostError = !!formik?.errors?.boost && formik?.errors?.boost !== ''
+    const hasForwardError = (!!formik?.errors?.forward?.[0].nym && formik?.errors?.forward?.[0].nym !== '') || (!!formik?.errors?.forward?.[0].pct && formik?.errors?.forward?.[0].pct !== '')
+    const hasMaxBidError = !!formik?.errors?.maxBid && formik?.errors?.maxBid !== ''
+
+    if (formik?.isSubmitting) {
+      setShow(hasForwardError || hasBoostError || hasMaxBidError ? formik?.isSubmitting : undefined)
+    }
+  }, [formik?.isSubmitting])
+
+  useEffect(() => {
+    if (show !== undefined && activeEventKey !== eventKey) {
+      decoratedOnClick()
+    }
+  }, [show])
+
```

Comment:
I think all of this logic should be in `adv-post-form.js` ... `AccordianItem` is used in many places throughout the code base, and we don't want this affecting unrelated code.

I could be missing something though.

Reply:
I had initially moved it because of what i viewed as a substaintially level of added complexity in https://github.com/stackernews/stacker.news/pull/1129/commits/ed3a958180d88c37588678ddf0a65523b10cfe83 vs just using the `hasForm` prop.  

Perhaps I could make it more clear about the useEffect dependencies not changing since `formik?` is null without the `hasForm` param `const formik = hasForm ? useFormikContext() : null`.  

Happy to move it back up one level to `adv-post-form` if you'd prefer that commit

Reply:
moved the logic in https://github.com/stackernews/stacker.news/pull/1129/commits/e4d273c19994af69999adbb561f9caa9fee1b460